lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_ansible_method_base.rb

```diff
@@ -49,17 +49,39 @@ def running_in_state_machine?
     def process_result(task)
       @aw.reload
       @workspace.update_workspace(@aw.output) if @aw.output
-      ansible_stats_from_task(task).each { |k, v| @workspace.persist_state_hash[k] = v }
+      ansible_stats_from_task(task).each do |k, v|
+        @workspace.persist_state_hash["#{ANSIBLE_STATS_PREFIX}#{k}"] = v
+        begin
+          update_object_with_stats(k, v)
+        rescue StandardError => err
+          $miq_ae_logger.error("Failed to update object with Ansible set_stats data [#{k} = #{v}]: #{err.message}")
+        end
+      end
       reset
     end
 
     def ansible_stats_from_task(task)
-      stats = task&.context_data&.dig(:ansible_runner_stdout, -1, 'event_data', 'artifact_data')
-      return {} unless stats
+      task&.context_data&.dig(:ansible_runner_stdout, -1, 'event_data', 'artifact_data') || {}
+    end
 
-      stats.each_with_object({}) do |(attr, val), obj|
-        obj["#{ANSIBLE_STATS_PREFIX}#{attr}"] = val
+    # key in the format of:
+    #   miq_provision__options__this__can__be__deep
+    #   miq_provision__status
+    def update_object_with_stats(key, value)
+      attrs = key.to_s.split("__")
```

Comment:
@lfu couldn't we prepend the key with ANSIBLE_STATS_PREFIX, instead of __

Reply:
@lfu Can we use :: instead of __
service_template_provision_task::options::config::xx::yy::zz = 1

Reply:
I would also prefer a different value than `__`.  Was curious why that one was chosen to begin with?  Does it have to do with that fact that most things are ENV vars?

Note sure on `::` specifically.  I like it, but other candidates are `:` or `/`.  In what context are these strings used, and are there any delimiters that are a hard restriction?

Reply:
From Ansible document - `Variable names should be letters, numbers, and underscores`. 

![Screen Shot 2019-11-04 at 09 53 30](https://user-images.githubusercontent.com/2593270/68130240-0074cb00-fee9-11e9-9545-2d850a5c2023.png)

Reply:
@mkanoor @Fryguy Are you okay with the "__" delimiter based on the above restrictions?

Reply:
👍

Reply:
👍